File: algorithms/2_medium/P/01010_pairs_of_songs_with_total_durations_divisible_by_60/solution.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        array<int, 60> freqs{};
        int res{ 0 };

        for (const auto& t : time) {
            const int curr{ t % 60 };
            const int other{ (60 - curr) % 60 };
            const auto freq_other{ freqs[static_cast<size_t>(other)]};

            res += freq_other;
            freqs[static_cast<size_t>(curr)] += 1;
        }

        return res;
    }
};

File: algorithms/2_medium/P/01010_pairs_of_songs_with_total_durations_divisible_by_60/solution.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        freqs = [0] * 60
        res = 0

        for t in time:
            curr = t % 60
            other = (60 - curr) % 60
            freq_other = freqs[other]

            res += freq_other
            freqs[curr] += 1

        return res
